addons/portal/static/tests/tours/portal.js

```diff
@@ -29,3 +29,89 @@ registry.category("web_tour.tours").add('portal_load_homepage', {
         }
     ]
 });
+
+var profilePictureSrc = "";
+
+registry.category("web_tour.tours").add("portal_profile_update", {
+    url: "/my/account",
+    steps: () => [
+        {
+            content: "Ensure that Profile Picture for Portal user is present",
+            trigger: ".o_portal_details .o_wportal_avatar_img",
+        },
+        {
+            content: "Ensure that Edit Button is present and Update the Profile Picture",
+            trigger: ".o_portal_details .o_portal_profile_pic_edit",
+            run: async function () {
+                {
+                    profilePictureSrc = document.querySelector(
+                        ".o_portal_details .o_wportal_avatar_img"
+                    ).src;
+                }
+                const fileInputEl = document.querySelector(".o_portal_file_upload");
+                // fetching image for upload
+                const response = await fetch("/portal/static/src/img/portal-addresses.svg");
+                const blob = await response.blob();
+
+                // simulating or mimicing file upload
+                const simulatedImageFile = new File([blob], "portal-addresses.png", {
+                    type: "image/png",
+                });
+                const dataTransfer = new DataTransfer();
+
+                dataTransfer.items.add(simulatedImageFile);
+                fileInputEl.files = dataTransfer.files;
+
+                fileInputEl.dispatchEvent(new Event("change", { bubbles: true, cancelable: true }));
+            },
+        },
+        {
+            content: "Click on save button",
+            trigger: ".o_portal_details button[type=submit]",
+            run: "click",
+        },
+        {
+            content: "Comparing previous Profile Picture with new One",
+            trigger: ".o_portal_contact_img",
+            run: function () {
+                const new_img = document.querySelector(".o_portal_contact_img");
+                if (new_img.src == { profilePictureSrc }) {
+                    throw new Error("Profile Picture is not updated!");
+                }
+                // updating profilePictureSrc with new picture to be used later.
+                {
+                    profilePictureSrc = new_img.src;
+                }
```

Comment:
no need to do this `(new_img.src == { profilePictureSrc })`  ==> do this instead `(new_img.src ===  profilePictureSrc)`.

compare it before saving else the value will always be ""  `profilePictureSrc` as the page will reload.